File: GymFit/GymFit.Core/Entities/Reservation.cs
namespace GymFit.Core
{
    public class Reservation:BaseEntity
    {
        public string Description { get; set; } = null!;
        public DateTime ReservationDate { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int Duration { get; set; }
        public int PauseDuration { get; set; }
        public int MaxCapacity { get; set; }
        public string DaysOfWeek { get; set; } = null!;
        public bool isUsed { get; set; }

        public int GymId { get; set; }
        public Gym Gym { get; set; } = null!;

        public int UserId { get; set; }
        public User User { get; set; } = null!;

        public int TrainerId { get; set; }
        public User Trainer { get; set; } = null!;

        public ICollection<Arrivals> Arrivals { get; set; } = null!;
        public ICollection<GroupReservations> GroupReservations { get; set; } = null!;
    }
}

File: SeminarskiGymFit/UI/gymfit_mobile/lib/models/reservation.dart
import 'package:gymfit_mobile/models/user.dart';

class Reservation {
  late int id;
  late String? description;
  late DateTime? ReservationDate;
  late DateTime? StartDate;
  late DateTime? EndDate;
  late int? Duration;
  late int? PauseDuration;
  late int? MaxCapacity;
  late String? DaysOfWeek;
  late bool? isUsed;
  late int? GymId;
  late User? user;
  late int? userId;
  late int? status;
  late User? Trainer;
  late int? trainerId;

  Reservation({
    required this.id,
    this.description,
    this.status,
    this.ReservationDate,
    this.StartDate,
    this.EndDate,
    this.Duration,
    this.PauseDuration,
    this.MaxCapacity,
    this.DaysOfWeek,
    this.GymId,
    this.Trainer,
    this.trainerId,
    this.isUsed,
    this.user,
    this.userId,
  });

  Reservation.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    userId = json['userId'];
    description = json['description'];
    ReservationDate = DateTime.parse(json['reservationDate']);
    StartDate = DateTime.parse(json['startDate']);
    EndDate = DateTime.parse(json['endDate']);
    Duration = json['duration'];
    PauseDuration = json['pauseDuration'];
    MaxCapacity = json['maxCapacity'];
    DaysOfWeek = json['daysOfWeek'];
    trainerId = json['trainerId'];
    GymId = json['gymId'];
    isUsed = json['isUsed'];
    status = json['status'];

   // user = User.fromJson(json['user']);
   Trainer = User.fromJson(json['trainer']);
  }

  Map<String, dynamic> toJson() {
  final Map<String, dynamic> data = <String, dynamic>{};
  data['id'] = id;
  data['userId'] = userId;
  data['description'] = description;

  return data;
}
}
